server/server_test.go

```diff
@@ -222,6 +222,31 @@ func TestOAuth2CodeFlow(t *testing.T) {
 			},
 		},
 		{
+			name: "verify at_hash",
+			handleToken: func(ctx context.Context, p *oidc.Provider, config *oauth2.Config, token *oauth2.Token) error {
+				rawIDToken, ok := token.Extra("id_token").(string)
+				if !ok {
+					return fmt.Errorf("no id token found")
+				}
+				idToken, err := p.Verifier().Verify(ctx, rawIDToken)
+				if err != nil {
+					return fmt.Errorf("failed to verify id token: %v", err)
+				}
+
+				var claims struct {
+					AtHash string `json:"at_hash"`
+				}
+				if err := idToken.Claims(&claims); err != nil {
+					return fmt.Errorf("failed to decode raw claims: %v", err)
+				}
+				if claims.AtHash == "" {
```

Comment:
The at_hash is used to verify that the ID Token <-> Access Token pair belongs together right? so will we have to modify the go-oidc Verify() function to check if received at_hash == expected at_hash?